content/module-reference/utility-modules/shared/tagging.md

```diff
@@ -2,29 +2,64 @@
-tags: 
+tags:
 view: lighttable, [darkroom], tethering, map
 ---
 
 Manage tags attached to images.
 
-Tags provide a means of adding information to images using a keyword dictionary. You can also manage tags as a hierarchical tree, which can be useful when their number becomes large. Only tags at the leaves of the tree add information to an image -- parent tags simply serve as categories and are are used to organise the leaf tags. For example, in the “color|white” tag, “color” is a category for the “white” tag.
+Tags provide a means of adding information to images using a keyword dictionary. You can also manage tags as a hierarchical tree, which can be useful when their number becomes large.
 
 Tags are physically stored in [XMP sidecar files](../../../overview/sidecar-files/_index.md) as well as in darktable's library database and can be included in [exported](./export.md) images.
 
+# definitions
+
+The following definitions assume that you have set up a single tag named `places|France|Nord|Lille`.
+
+tag
+: A tag is a descriptive string that may be attached to an image. A tag can be a single term or a sequence of connected terms forming a path, separated by the pipe symbol. For example, `places|France|Nord|Lille` defines a single tag, with each term in the path forming a subset of those before it. You can attach as many tags to an image as you like.
+
+You can assign properties (name, private, category, synonyms and image order) to a tag.
+
+node
+: Any path that forms part of a tag is a node. In the above example, `places`, `places|France`, `places|France|Nord` and `places|France|Nord|Lille` are all nodes.
+
+free node
+: Any node that is not explicitly defined as a tag is called a free node. In the above example, `places`, `places|France` and `places|France|Nord` are all free nodes. You cannot set any properties, except “name”, for a free node and you cannot add a free node to an image. See the “multiple tags” section below for more information.
+
+category
+: Any tag can be flagged by the user as being a “category”. All free nodes are implicitly defined as “category” tags.
+
+# multiple tags
+
+The above definitions considered a simple example – a single tag and its properties. Consider instead the scenario where the following tags are each separately defined in darktable. For now, assume that none of the tags has been flagged as a “category”.
+
+`places|France|Nord|Lille`
+`places|France|Nord`
+`places|France`
+`places|England|London`
+
+In this case, each of these pipe-delimited strings is a tag. The nodes are `places`, `places|France`, `places|France|Nord`, `places|France|Nord|Lille`, `places|England` and `places|England|London`.
+
+The only free nodes are `places` and `places|England`. Both of these free nodes are also (by implication) categories.
+
+You can attach any of these tags to any image. Any tags attached to an image (except “category” tags) can be included when that image is exported (see [export](./export.md) module).
```

Comment:
I would say 'tags flagged as "category"' instead of '"category" tags'

Reply:
Actually I was going to suggest going the other way. "tags flagged as "category"' is pretty ugly English. Given that we've defined category tags in the first section I'd say "category tags" is a neater phrase to use throughout. As long as we're consistent and have defined our terms well it should be fine.